### PR TITLE
Migrate plugin to build process + preact frontend

### LINKED ISSUE #2: Nice lib!
Hey,

Nice lib!

I saw your video on YouTube whilst looking to see how folks do their Tailwind stuff.

This looks like a solid start, but I''d be happy to contribute to help make the code more maintainable, and even add a few features. I'm also contributing to [FigmaToCode](https://github.com/bernaferrari/FigmaToCode/issues?q=involves%3Adavestewart+) at the moment, so I think there could be some nice tie-ins.

Are you up to speed on pull requests?

### PR DESCRIPTION
**Pull request by davestewart**
Hey,

It ended up being a pretty chunky update!

As I mentioned in the issue, in order to consolidate the config, we needed a build process. And if we're doing a build process, it makes sense to use something robust, so I chose the Create Figma Plugin platform. It's Preact (a simpler React) but the great advantage is that they've taken the time to rebuild the [Figma UI as components](https://yuanqing.github.io/create-figma-plugin/storybook/?path=/story/index--index), which makes it super easy to just build something that looks like Native Figma.

Once the UI was in, it was pretty much a case of simplifying the existing code to make it more generic.

The final feature set is:

- Native and consistent-looking UI
- Groups and values for for Units, Fonts, Colors
- Variable names, values, scopes and code syntax (used for plugins, and the main feature I need)

splash

As mentioned, it was a bit of an experiment for me, but thought it would be both fun and useful.

Hopefully you'll accept this PR even though it sound like the code is possibly a few levels up from what you're used to. I'm happy to answer any questions you may have!

A few things stillI could have done, but not required:

- could [add Tailwind](https://yuanqing.github.io/create-figma-plugin/ui/#using-tailwind-css) to make the styling simpler
- review TS type generics for config / chosen user options

Closes #2

### CODE REVIEW

**Comment by masha312** [comment]
This is so cool, thank you! Will try to take a look in the next couple of days 🙏

File: README.md [hunk 1]
```diff
@@ -1 +1,98 @@
...
+
+### Usage
+
+You need to [build](#build-the-plugin) or [install](https://www.figma.com/community/plugin/1281668905420544225/variables-starter) the plugin before using.
+
+Then:
+
+- go Toolbar > Resources > Plugins > Variables Boilerplate
+- click to toggle individual options
+- click "Toggle All" to toggle an entire group
```
Comment (masha312) [inline README.md]:
```suggestion
- click "Toggle all" to toggle an entire group
```

File: src/app/components/Fieldset.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if (onToggle) {
+      onToggle()
+    }
+  }
+
+  return <div>
+    <VerticalSpace space="extraLarge" />
+    <Text>
+      <Muted>{label}
+        {onToggle && <span> | <Link href="#" onClick={onClick}>Toggle All</Link></span>}
```
Comment (masha312) [inline src/app/components/Fieldset.tsx]:
```suggestion
        {onToggle && <span> | <Link href="#" onClick={onClick}>Toggle all</Link></span>}
```

**Review by masha312 — APPROVED** [review]
Looks amazing to me, thank you so much :) 
I only wish Figma allowed periods in variable names.. 

Tiny thing: do you mind if we do sentence case for "Toggle all"?

For the future: I think loading state could be useful here, especially when it's creating **all** variables. Something I could add later though.

Lastly, please let me know your Figma username if you'd like to be added as a co-creator there 🙏

File: README.md (hunk 1 above)
Comment (davestewart) [reply README.md]: I agree!

File: src/app/components/Fieldset.tsx (hunk 2 above)
Comment (davestewart) [reply src/app/components/Fieldset.tsx]: ✅

**Comment by davestewart** [comment]
OK, changes made.

> **EDIT:** Eeeek! It looks like there's an error when there are no variable collections. Let me fix that and come back.

PS. How long does it take to create the variables on your machine?

And, here I am on Figma: https://www.figma.com/@davestewart

Thanks!

**Comment by davestewart** [comment]
OK, collection bug fixed.

> For the future: I think loading state could be useful here, especially when it's creating all variables. Something I could add later though.

FWIW, the time to create variables for me is almost instant.

Possibly there could be some user feedback, such as a toast reporting how many variables were created.

Regarding the loading spinner / reporting, we would need to implement [message passing](https://www.figma.com/plugin-docs/api/properties/figma-ui-postmessage/). The framework already has some `emit()` code (essentially, sending a message) but it doesn't look like it returns a value, so I don't think the messaging is two-way.

Ideally the flow in the UI would be something like:

```ts
setLoadingState(true)
const result: { message: string, numCreated: number } = await emit('CREATE', state)
setLoadingState(false)
if (result.message) {
	setMessage(result.message)
    setTimeout(() => setMessage(), 3000)
}
```

Happy to help you when the time comes, or I can leave it up to a challenge for you!

Anyway. All ready to go now. 🚀

**Comment by davestewart** [comment]
Yay!

Don't forget to publish.

I can't remember what the process is for Figma plugins, or what files need to be included or how.

The way the build is set up with the framework, the build folder does _not_ contain the manifest. You can modify this by setting a custom `--output` folder, but it seems this [buggers the watch process](https://github.com/yuanqing/create-figma-plugin/issues/237).

Anyway, any issues, I'll be around.

**Comment by masha312** [comment]
Should be live ✔️

**Pull request merged**